coverage/src/lib.rs

```diff
@@ -0,0 +1,302 @@
+// Licensed under the Apache-2.0 license
+
+use anyhow::Context;
+use bit_vec::BitVec;
+use caliptra_builder::{build_firmware_elf, FwId, SymbolType};
+use elf::endian::AnyEndian;
+use elf::ElfBytes;
+use std::collections::hash_map::{DefaultHasher, Entry};
+use std::collections::{HashMap, HashSet};
+use std::fs::File;
+use std::hash::Hasher;
+use std::io::{BufRead, BufReader, BufWriter, Write};
+use std::path::{Path, PathBuf};
+
+pub const CPTRA_COVERAGE_PATH: &str = "CPTRA_COVERAGE_PATH";
+
+pub struct CoverageMap {
+    pub map: HashMap<u64, BitVec>,
+}
+
+impl CoverageMap {
+    pub fn new(paths: Vec<PathBuf>) -> Self {
+        let mut map = HashMap::<u64, BitVec>::default();
+        for path in paths {
+            let new_entry = get_entry_from_path(&path);
+            match map.entry(new_entry.0) {
+                Entry::Vacant(e) => {
+                    e.insert(new_entry.1);
+                }
+                Entry::Occupied(mut e) => {
+                    e.get_mut().or(&new_entry.1);
+                }
+            }
+        }
+        Self { map }
+    }
+}
+pub struct CoverageMapEntry(u64, BitVec);
+pub fn get_entry_from_path(path: &PathBuf) -> CoverageMapEntry {
+    let filename = path.file_name().unwrap().to_str().unwrap();
+    let tag = filename
+        .split('-')
+        .nth(1)
+        .unwrap()
```

Comment:
nitpick: panicking where this should probably be an error

Reply:
follow up pr.